shared/reducers/fs.js

```diff
@@ -102,6 +110,47 @@ const updatePathItem = (oldPathItem?: ?Types.PathItem, newPathItem: Types.PathIt
   }
 }
 
+const updateTlf = (oldTlf?: ?Types.Tlf, newTlf: Types.Tlf): Types.Tlf => {
+  if (!oldTlf) {
+    return newTlf
+  }
+  // TODO: Ideally this should come in with other data from the same RPC.
+  const newTlfDontClearSyncConfig = newTlf.syncConfig ? newTlf : newTlf.set('syncConfig', oldTlf.syncConfig)
+  if (!newTlf.resetParticipants.equals(oldTlf.resetParticipants)) {
+    return newTlfDontClearSyncConfig
+  }
+  if (!I.is(newTlfDontClearSyncConfig.waitingForParticipantUnlock, oldTlf.waitingForParticipantUnlock)) {
+    return newTlfDontClearSyncConfig
+  }
+  if (!I.is(newTlfDontClearSyncConfig.youCanUnlock, oldTlf.youCanUnlock)) {
+    return newTlfDontClearSyncConfig
+  }
+  if (
+    !I.is(newTlfDontClearSyncConfig.syncConfig, oldTlf.syncConfig) &&
+    !(
+      newTlfDontClearSyncConfig.syncConfig &&
+      oldTlf.syncConfig &&
+      newTlfDontClearSyncConfig.syncConfig.mode === 'partial' &&
+      oldTlf.syncConfig.mode === 'partial' &&
+      newTlfDontClearSyncConfig.syncConfig.enabledPaths.equals(oldTlf.syncConfig.enabledPaths)
```

Comment:
This is extremely difficult to decipher. I'd prefer DeMorgan'ing this:
```javascript
if (
  !I.is(...) && (
    !newTlfDontClearSyncConfig.syncConfig ||
    !oldTlf.syncConfig ||
    !newTlfDontClearSyncConfig.syncConfig.mode === 'partial' ||
    !oldTlf.syncConfig.mode === 'partial' ||
    !newTlfDontClearSyncConfig.syncConfig.enabledPaths.equals(oldTlf.syncConfig.enabledPaths)
  )
)
```

However, it might be better to make a helper function to do this check, that explains what it's meant to represent.